src/main/java/com/auction/auction/bid/controller/UserController.java

```diff
@@ -1,52 +1,46 @@
 package com.auction.auction.bid.controller;
 
 import com.auction.auction.bid.dto.UserDTO;
+import com.auction.auction.bid.interfaces.UserInterface;
 import com.auction.auction.bid.model.User;
 import com.auction.auction.bid.service.UserService;
 import jakarta.validation.Valid;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-
 import java.util.List;
 
 @RestController
 @RequestMapping("/users")
 public class UserController {
 
     @Autowired
-    private final UserService userService;
+    private UserInterface userService;
 
-    public UserController(UserService userService) {
-        this.userService = userService;
-    }
 
     @PostMapping("/users")
     public ResponseEntity<User> createUser(@RequestBody @Valid UserDTO userDTO) {
-        User user = new User(
-                userDTO.getUsername(),
-                userDTO.getPassword(),
-                userDTO.getEmail()
-        );
-        User createdUser = userService.createUser(user);
-        return ResponseEntity.status(HttpStatus.CREATED).body(createdUser);
+        User response = userService.createUser(userDTO);
+        return ResponseEntity.status(HttpStatus.CREATED).body(response);
+    }
+
+    @PutMapping("/{id}")
```

Comment:
your User object already have Id in it so you shoud not sent additional parameter "id". All you need is just User object and PutMapping without any values in it